lib/graphql-hive/usage_reporter.rb

```diff
@@ -21,13 +20,15 @@ def initialize(options, client)
         @client = client
         @options_mutex = Mutex.new
         @sampler = Sampler.new(options[:collect_usage_sampling], options[:logger]) # NOTE: logs for deprecated field
-        @queue = BoundedQueue.new(bound: options[:buffer_size], logger: options[:logger])
+        @queue = Thread::SizedQueue.new(options[:queue_size])
 
         start_thread
       end
 
       def add_operation(operation)
-        @queue.push(operation)
+        @queue.push(operation, true)
+      rescue ThreadError
+        @options[:logger].error("SizedQueue is full, discarding operation. Size: #{@queue.size}, Max: #{@queue.max}")
```

Comment:
```suggestion
        @options[:logger].error("[hive] SizedQueue is full, discarding operation. Size: #{@queue.size}, Max: #{@queue.max}")
```

One small change because it's hard to tell from the logs that this is coming from Hive.

Reply:
This is now in sync code along with requests, and not in the thread. So it might be OK at this point. To research!